src/main/java/io/fabric8/maven/docker/config/handler/compose/DockerComposeServiceWrapper.java

```diff
@@ -396,4 +406,97 @@ private Arguments asArguments(Object command, String label) {
     private void throwIllegalArgumentException(String msg) {
         throw new IllegalArgumentException(String.format("%s: %s - ", composeFile, name) + msg);
     }
+
+    /**
+     * Examines the supplied {@code volume} string, and resolves relative paths.  The returned string is guaranteed to
+     * be absolute, but <em>not</em> guaranteed to exist.
+     * <p>
+     * Volumes may be defined inside of {@code service} blocks <a href="https://docs.docker.com/compose/compose-file/compose-file-v2/#volumes-volume_driver">
+     * as documented here</a>:
+     * </p>
+     * <pre>
+     * volumes:
+     * # Just specify a path and let the Engine create a volume
+     * - /var/lib/mysql
+     *
+     * # Specify an absolute path mapping
+     * - /opt/data:/var/lib/mysql
+     *
+     * # Path on the host, relative to the Compose file
+     * - ./cache:/tmp/cache
+     *
+     * # User-relative path
+     * - ~/configs:/etc/configs/:ro
+     *
+     * # Named volume
+     * - datavolume:/var/lib/mysql"
+     * </pre>
+     * <p>
+     * This method only operates on volume strings that are relative: beginning with {@code ./} or {@code ~/}.  Relative
+     * paths beginning with {@code ./} are absolutized relative to the supplied base directory, which <em>must</em> be
+     * absolute.  Paths beginning with {@code ~/} are interpreted relative to {@code new File(System.getProperty(
+     * "user.home"))}, and {@code baseDir} is ignored.
+     * </p>
+     * <p>
+     * Volume strings that do not begin with a {@code ./} or {@code ~/} are returned as-is.
+     * </p>
+     * <h3>Examples:</h3>
+     * <p>
+     * Given {@code baseDir} equal to "/path/to/basedir" and a {@code volume} string equal to
+     * "./reldir:/some/other/dir", this method returns {@code /path/to/basedir/reldir:/some/other/dir}
+     * </p>
+     * <p>
+     * Given {@code baseDir} equal to "/path/to/basedir" and a {@code volume} string equal to
+     * "~/reldir:/some/other/dir", this method returns {@code /path/to/user/home/reldir:/some/other/dir}
+     * </p>
+     *
+     * @param baseDir the base directory used to resolve paths beginning with {@code ./}; <em>must</em> be absolute
+     * @param volume the volume string from the docker-compose file
+     * @return the volume string, with any relative paths resolved as absolute paths
+     */
+    static String resolveRelativeVolumeBinding(File baseDir, String volume) {
```

Comment:
It would be awesome if we could use the same kind of parsing for the regular configuration, too. 

E.g. in 

* https://github.com/rhuss/docker-maven-plugin/blob/674e80fdf56d31fdd263366d53e7a5cf0dc064b3/src/main/java/io/fabric8/maven/docker/service/RunService.java#L258
* https://github.com/rhuss/docker-maven-plugin/blob/674e80fdf56d31fdd263366d53e7a5cf0dc064b3/src/main/java/io/fabric8/maven/docker/service/RunService.java#L332

The challenge there is to get the basedir into the method (which must not have a reference to any Maven code). Needs probably given through top-down.